resources/js/components/ExistingMedia.vue

```diff
@@ -0,0 +1,165 @@
+<template>
+  <!-- Modal -->
+  <div class="fixed pin-l pin-t p-8 h-full w-full z-50" :class="{'hidden': !open, 'flex': open}">
+
+    <!-- Modal Background -->
+    <div class="absolute bg-black opacity-75 pin-l pin-t h-full w-full" v-on:click="close"></div>
+
+    <!-- Modal Content -->
+    <div class="flex flex-col bg-white p-4 h-full relative w-full">
+      <!-- Header bar -->
+      <div class="border-b border-40 pb-4 mb-4">
+        <div class="flex -mx-4">
+          <!-- Heading -->
+          <div class="px-4 self-center">
+            <h3>Existing Media</h3>
+          </div>
+
+          <!-- Search -->
+          <div class="px-4 self-center">
+            <div class="relative">
+              <icon type="search" class="absolute search-icon-center ml-3 text-70" />
+              <input type="search"
+                     placeholder="Search by name or file name"
+                     class="pl-search form-control form-input form-input-bordered w-full"
+                     v-model="requestParams.search_text"
+                     @input="search"
+                     @change="search"
+                     @keydown.enter.prevent="search">
+            </div>
+          </div>
+
+          <!-- Close -->
+          <div class="px-4 ml-auto self-center">
+            <button type="button" class="form-file-btn btn btn-default btn-primary" v-on:click="close">Close</button>
+          </div>
+        </div>
+      </div>
+
+
+      <div class="flex-grow overflow-x-hidden overflow-y-scroll">
+        <!-- When we have results show them -->
+        <template v-if="data.length > 0">
+          <div class="flex flex-wrap -mx-4 -mb-8">
+            <template v-for="(item, key) in data">
+              <existing-media-item :item="item" :key="key" v-on:select="$emit('select', item) && close()"></existing-media-item>
+            </template>
+          </div>
+        </template>
+
+        <!-- Show "Loading" or "No Results Found" text -->
+        <template v-if="loading">
+          <h4 class="text-center m-8">Loading...</h4>
+        </template>
+        <template v-else-if="data.length == 0">
+          <h4 class="text-center m-8">No results found</h4>
+        </template>
+      </div>
+
+      <!-- Next page -->
+      <template v-if="showNextPage">
+        <div class="flex-shrink border-t border-40 pt-4 mt-4 text-right">
+          <button type="button" class="form-file-btn btn btn-default btn-primary ml-auto" v-on:click="nextPage">Load Next Page</button>
+        </div>
+      </template>
+    </div>
+  </div>
+</template>
+
+<script>
+  import ExistingMediaItem from './ExistingMediaItem';
+  import debounce from 'lodash/debounce';
+
+  export default{
+    components: {
+      ExistingMediaItem
+    },
+    data () {
+      let aThis = this;
+      return {
+        requestParams: {
+          search_text: '',
+          page: 1,
+          per_page: 18
+        },
+        data: [],
```

Comment:
this makes it a bit confusion using data inside data. i'd suggest to use `items` or something similar.